src/cozmo/robot.py

```diff
@@ -130,6 +131,55 @@ def _encode(self):
                                                 useManualSpeed=False,
                                                 usePreDockPose=False)
 
+class DockWithCube(action.Action):
+    '''Represents the dock with cube action in progress.
+
+    Returned by :meth:`~cozmo.robot.Robot.dock_with_cube`
+    '''
+    _action_type = _clad_to_engine_cozmo.RobotActionType.ALIGN_WITH_OBJECT
+
+    def __init__(self, obj, **kw):
+        super().__init__(**kw)
+        #: The object (e.g. an instance of :class:`cozmo.objects.LightCube`) that is being put down
+        self.obj = obj
+
+    def _repr_values(self):
+        return "object=%s" % (self.obj)
+
+    def _encode(self):
+        return _clad_to_engine_iface.AlignWithObject(objectID=self.obj.object_id,
+                                                     distanceFromMarker_mm=util.distance_mm(0).distance_mm,
+                                                     approachAngle_rad=0, #I've chosen not to expose the approach angle, as it doesn't seem to have any effect
+                                                     alignmentType=alignment.AlignmentTypes.Body.id,
+                                                     useApproachAngle=False,
+                                                     usePreDockPose=False,
+                                                     useManualSpeed=False)
+
+class RollCube(action.Action):
+    '''Represents the roll cube action in progress.
+
+    Returned by :meth:`~cozmo.robot.Robot.roll_cube`
+    '''
+
+    _action_type = _clad_to_engine_cozmo.RobotActionType.ROLL_OBJECT_LOW
+
+    def __init__(self, obj, check_for_object_on_top, **kw):
+        super().__init__(**kw)
+        #: The object (e.g. an instance of :class:`cozmo.objects.LightCube`) that is being put down
+        self.obj = obj
+        #: bool: whether to check if there is an object on top
+        self.check_for_object_on_top = check_for_object_on_top
+
+    def _repr_values(self):
+        return "object=%s, check_for_object_on_top=%s" % (self.obj, self.check_for_object_on_top)
+
+    def _encode(self):
+        return _clad_to_engine_iface.RollObject(objectID=self.obj.object_id,
+                                                approachAngle_rad = 0, #I've chosen not to expose the approach angle, as it doesn't seem to have any effect
```

Comment:
I spent some time trying to mess with approach angle with the flag set to true and couldn't figure out what effect it had.  My assumption was that it would make cozmo drive around to another side of the block, and roll it that way.  Changing the comment, though.